opentelemetry-sdk/tests/metrics/test_metrics.py

```diff
@@ -96,12 +96,56 @@ def test_get_handle(self):
             self.assertEqual(metric.handles.get(label_values), handle)
 
 
-class TestCounterHandle(unittest.TestCase):
+class TestCounter(unittest.TestCase):
+    def test_add(self):
+        metric = metrics.Counter("name", "desc", "unit", int, ("key",))
+        handle = metric.get_handle(("value",))
+        metric.add(("value",), 3)
+        metric.add(("value",), 2)
+        self.assertEqual(handle.data, 5)
+
     def test_update(self):
-        handle = metrics.CounterHandle(float, True, False)
-        handle.update(2.0)
-        self.assertEqual(handle.data, 2.0)
+        metric = metrics.Counter("name", "desc", "unit", int, ("key",))
+        handle = metric.get_handle(("value",))
+        metric.update(("value",), 3)
+        metric.update(("value",), 2)
+        self.assertEqual(handle.data, 5)
+
+
+class TestGauge(unittest.TestCase):
+    def test_set(self):
+        metric = metrics.Gauge("name", "desc", "unit", int, ("key",))
+        handle = metric.get_handle(("value",))
+        metric.set(("value",), 3)
```

Comment:
maybe for a future PR: looking at the prometheus API you can actually call out the tag you want to set by name:

https://github.com/prometheus/client_python#labels

Which, I understand is less performant, but is easier to read and understand. Maybe worth a discussion to see if we should support them?

I've also realized while typing this that most invocations won't really need this since they'll be using variables that probably state the value just fine:

`metric.set((environment,), 3)`

Reply:
That's a good suggestion. Part of the API specs is actually implementing the LabelSet(s) as a first class API concept. LabelSets can be created and then passed into methods `add`, `set`, and `record`.

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics-user.md#label-set-calling-convention